gpflow/models/model.py

```diff
@@ -57,8 +52,29 @@ def log_prior_density(self) -> tf.Tensor:
         else:
             return to_default_float(0.0)
 
+    def log_posterior_density(self, *args, **kwargs) -> tf.Tensor:
+        """
+        This may be the posterior with respect to the hyperparameters (e.g. for
+        GPR) or the posterior with respect to the function (e.g. for GPMC and
+        SGPMC). It assumes that maximum_likelihood_objective() is defined
+        sensibly.
+        """
+        return self.maximum_likelihood_objective(*args, **kwargs) + self.log_prior_density()
+
+    def maximum_a_posteriori_objective(self, *args, **kwargs) -> tf.Tensor:
+        """
+        Maximum a-posteriori objective. Assumes that maximum_likelihood_objective() is the log marginal likelihood and adds the log density of all priors.
```

Comment:
you say "assumes". what happens when it's not e.g. when it's the elbo?

Reply:
that's fine - I was thinking more "it has to be the *log* marginal likelihood not the non-logged actual marginal likelihood"